src/core/http-client.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpRequest, HttpResponse } from "./http";
+import { HttpHandler } from "./http4ts";
+import { stringBody } from "../node";
+
+type Headers = {
+  forEach: (cb: (value: string, key: string) => void) => void;
+};
+
+export type Fetch = (
+  url: string,
+  init?: { method: string; body: string | undefined }
+) => Promise<{
+  headers: Headers;
+  text: () => Promise<string>;
+  status: number;
+}>;
+
+export function createHttpClient(theFetch: Fetch): HttpHandler {
```

Comment:
Limiting the client to use something that looks exactly like Fetch seems like a constraint. For instance if we want to provide Axios here, we should create an adapter that translates Axios to the type of Fetch. Fetch return type is also not exactly the same as HttpResponse, so the client is not very symmetric. If you want to provide a handler that is FetchClientHandler, then it is a different story, but the generic client HTTP handler is too specific in this case.

Reply:
We have to come up with an interface in the end. No matter using fetch or axios or something that we build ourselves we will limit the parameter of this function to that interface. Fetch is probably the best because it is going to be the javascript standard. Although we may need to later build httpClients that accept other popular client libraries.

> Fetch return type is also not exactly the same as HttpResponse, so the client is not very symmetric.

The user of httpClient does not use fetch. The result of `createHttpClient` is an `HttpHandler` which is symmetric (see `simple-client.ts` for usage).

We don't want to provide `FetchHttpHandler`, but a generic client which corresponds with `HttpHandler` interface.